.github/workflows/pr-preview.yml

```diff
@@ -17,6 +17,10 @@ jobs:
   upload-whl:
     runs-on: ubuntu-latest
 
+    permissions:
+      contents: write # This is required for actions/checkout
+      statuses: write # This is required for pbrandone/create-status-action
```

Comment:
When you say `actions/checkout` requires write content permissions - is this something we are missing across the other workflows?

Reply:
read permission should be enough for our user case.